t/data_migration_pipeline/main.go

```diff
@@ -52,37 +59,58 @@ func parseDataModel(record []string) *DataModel {
 }
 
 // emitResult emits data models to be written to Spanner
-func emitResult(s beam.Scope, lines beam.PCollection) beam.PCollection {
+func emitResult(ctx context.Context, s beam.Scope, lines beam.PCollection) beam.PCollection {
 	dataModels := beam.ParDo(s, func(line string, emit func(*DataModel)) {
 		reader := csv.NewReader(strings.NewReader(line))
 		csvLine, err := reader.Read()
 		if err != nil {
 			log.Fatalf("Failed to read record: %v", err)
 		}
 		emit(parseDataModel(csvLine))
+		count.Inc(ctx, 1)
 	}, lines)
+
 	return dataModels
 }
 
 func main() {
+	// Handle context cancellation.
+	ctx, cancel := signal.NotifyContext(context.Background(),
+		syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	// Set up metric
+	ctx = metrics.SetPTransformID(ctx, "pipeline")
```

Comment:
Any reason you set SetPTransformID? This is a simple one-step pipeline and we dont necessarily need to distinguish different PTransforms in the metrics. Also if you want to have this, a PTransform id should be clear which tranform stage it is.

Reply:
if I don't add it, it gives `Unable to dump metrics: provided context doesn't contain metrics Store.` Following the example https://pkg.go.dev/github.com/apache/beam/sdks/v2/go/pkg/beam#example-package-MetricsDeclaredAnywhere Not sure whether it's a good example

Reply:
Got that. I might missed sth, fond this example not setting PTransformID: https://github.com/spotify/beam/blob/bddf2be8057fd2f7966451b8a4a4a90d8ff02820/sdks/go/test/load/pardo/pardo.go#L103

You can also set that, I'll defer to your choice, if so, do 
ctx = metrics.SetPTransformID(ctx, "emitResult")

Reply:
my 2 cents after reading the code: in summary, the mechanism are different. Actually `SetPTransformID` is nothing related to `RunWithMetrics`. `DumpToLog` must have `SetPTransformID` to setup store and based on it to find counter stored in context. `RunWithMetrics` requires ParDo with counter before hand. 
I guess print out metrics is different in local run once and streaming? Another example using a total different solution https://beam.apache.org/documentation/programming-guide/#using-metrics

Reply:
The other thing is to keep the case simple, I don't want to overkill the case

Reply:
I see, if it's required by DumpToLog, lets go with this solution.